crawler4j/src/main/java/edu/uci/ics/crawler4j/crawler/WebCrawler.java

```diff
@@ -583,6 +551,98 @@ private void processPage(WebURL curURL) throws IOException, InterruptedException
+    		throws IOException, InterruptedException {
+    	ParseData parseData = page.getParseData();
+        List<WebURL> toSchedule = new ArrayList<>();
+        int maxCrawlDepth = myController.getConfig().getMaxDepthOfCrawling();
+        for (WebURL webURL : parseData.getOutgoingUrls()) {
+            webURL.setParentDocid(curURL.getDocid());
+            webURL.setParentUrl(curURL.getURL());
+            int newdocid = docIdServer.getDocId(webURL.getURL());
+            if (newdocid > 0) {
+                // This is not the first time that this Url is visited. So, we set the
+                // depth to a negative number.
+                webURL.setDepth((short) -1);
+                webURL.setDocid(newdocid);
+            } else {
+                webURL.setDocid(-1);
+                webURL.setDepth((short) (curURL.getDepth() + 1));
+                if ((maxCrawlDepth == -1) || (curURL.getDepth() < maxCrawlDepth)) {
+                    if (shouldVisit(page, webURL)) {
+                        if (robotstxtServer.allows(webURL)) {
+                            webURL.setDocid(docIdServer.getNewDocID(webURL.getURL()));
+                            toSchedule.add(webURL);
+                        } else {
+                            logger.debug(
+                                "Not visiting: {} as per the server's \"robots.txt\" " +
+                                "policy", webURL.getURL());
+                        }
+                    } else {
+                        logger.debug(
+                            "Not visiting: {} as per your \"shouldVisit\" policy",
+                            webURL.getURL());
+                    }
+                } else {
+                	onMaxDepthReached(page, webURL);
+                }
+            }
+        }
+        frontier.scheduleAll(toSchedule);
+    }
+    
+    /**
+     * Adds the redirection destiny to the scheduler.
+     * 
+     * @param page 
+     * 		the page object that is just fetched and parsed.
+     * @param curURL
+     * 		current URL
+     * @param movedToUrl
+     * 		destiny URL
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    protected void followRedirection(Page page, WebURL curURL, String movedToUrl) 
+    		throws IOException, InterruptedException {
+    	int newDocId = docIdServer.getDocId(movedToUrl);
+        if (newDocId > 0) {
+            logger.debug("Redirect page: {} is already seen", curURL);
+            return;
+        }
+
+        WebURL webURL = new WebURL();
+        webURL.setTldList(myController.getTldList());
+        webURL.setURL(movedToUrl);
+        webURL.setParentDocid(curURL.getParentDocid());
+        webURL.setParentUrl(curURL.getParentUrl());
+        webURL.setDepth(curURL.getDepth());
+        webURL.setDocid(-1);
+        webURL.setAnchor(curURL.getAnchor());
+        if (shouldVisit(page, webURL)) {
+            if (shouldFollowLinksIn(webURL) && robotstxtServer.allows(webURL)) {
```

Comment:
This line was
if (!shouldFollowLinksIn(webURL) || robotstxtServer.allows(webURL))

I believe it was a bug, as the log seems completly opposite to expected behaviour